build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/rest/compat/RestCompatTestTransformTask.java

```diff
@@ -112,6 +124,28 @@ private static boolean doesNotHaveCatOperation(ObjectNode doNodeValue) {
         return true;
     }
 
+    public void skipTest(String testName, String reason) {
+        //The tests are defined by 3 parts a/b/c where
```

Comment:
This whole thing is a bit brittle and won't work in some cases we have, like when we have nested folders, in which case segments 1 and 2 would be "folders" and segment 3 would be the file name. It might be simpler to not try and do any kind of parsing here and instead down below where we actually apply the filters we just match prefixes. That is, if we are currently iterating over the file `monitoring/bulk/10_basic.yml` and we find a find a skip where the test name is `monitoring/bulk/10_basic/foo/bar` we construct a skip assuming `foo/bar` is the test name. This also avoid special handling for test names with slashes, we just assume everything after the file name is the test name.

Essentially, instead of trying to _infer_ the file from the test name, we simply check against known files at transform time. We can also implement a check where if have skip entries that don't match any of the input files we throw an error.

Reply:
I updated the code to be more robust w.r.t. number of forward slashes, similar (but not exact) to your recommendation. thanks!

Reply:
Also, the check to throw an error on skips not implemented is a bit tricky and felt a bit of overkill here since the only reason you want to skip is to avoid an error, so if you get it wrong you will know. The case where it would help is if you remove a test entirely and don't want the bloat of the additional configuration. I suspect this is pretty rare, but can add in the check if you think it is worth the extra lines of code/tracking.